lib/Net/Async/Redis.pm

```diff
@@ -1096,6 +1096,48 @@ sub execute_command {
      ->retain;
 }
 
+=head2 RESP3 and RESP2 compatibility
+
+In RESP3 some of the responses are structured differently from RESP2 
+L<Net::Async::Redis> guarantees the same structure unless you have explicitly requested the new types
+
+=cut
+
+around [qw(xread xreadgroup)] => async sub {
+    my ($code, $self, @args) = @_;
+    return await $self->$code(@args) if $self->{hashrefs};
+
+    my $response = await $self->$code(@args);
+
+    # protocol_level is detected while connecting checking before this point is wrong.
+    return $response if $self->{protocol_level} eq 'resp2';
+
+    my $compatible_response = [];
+    my %response_hash = $response->@*;
+    for my $stream (keys %response_hash) {
+        push $compatible_response->@*, [$stream, $response_hash{$stream}];
+    }
```

Comment:
im not sure if we want to have `use List::Util qw(pairmap);` here, but if we do this can be
```suggestion
    my $compatible_response = [ pairmap { [ $a, $b ] } $response->@* ];
```